src/config/plugin.ts

```diff
@@ -41,15 +41,34 @@ const search = (cmd: any) => {
   return Object.values(cmd).find((cmd: any) => typeof cmd.run === 'function')
 }
 
+const GLOB_PATTERNS = [
+  '**/*.+(js|cjs|mjs|ts|tsx|mts|cts)',
+  '!**/*.+(d.ts|test.ts|test.js|spec.ts|spec.js|d.mts|d.cts)?(x)',
+]
+
+function processCommandIds(files: string[]): string[] {
+  return files.map((file) => {
+    const p = parse(file)
+    const topics = p.dir.split('/')
+    const command = p.name !== 'index' && p.name
+    const id = [...topics, command].filter(Boolean).join(':')
+    return id === '' ? '.' : id
+  })
+}
+
 export class Plugin implements IPlugin {
   alias!: string
 
   alreadyLoaded = false
 
   children: Plugin[] = []
 
+  commandIDs!: string[]
```

Comment:
doesn't the code below break the interface?  When it was a getter, it would always return something.  The `!` means that it's not possible to have a Plugin (class) without commandIDs, but without calling `load()` you would.

Maybe init them as empty arrays and then load can fix them if it runs?